modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -432,4 +436,280 @@ public static ReadOnlyBooleanProperty accessibilityActiveProperty() {
         }
         return accessibilityActiveProperty.getReadOnlyProperty();
     }
+
+    /**
+     * Gets the preferences of the current platform.
+     * <p>
+     * The map returned from this method is unmodifiable, which means that keys and values cannot
+     * be added, removed, or updated. Calling any mutator method on the map will always cause
+     * {@code UnsupportedOperationException} to be thrown. However, the mappings will be updated
+     * by JavaFX when the operating system reports that a platform preference has changed.
+     *
+     * @return the {@code Preferences} instance
+     * @since 22
+     */
+    public static Preferences getPreferences() {
+        return PlatformImpl.getPlatformPreferences();
+    }
+
+    /**
+     * Contains UI preferences of the current platform.
+     * <p>
+     * {@code Preferences} extends {@link ObservableMap} to expose platform preferences as key-value pairs.
+     * For convenience, {@link #getInteger}, {@link #getDouble}, {@link #getBoolean}, {@link #getString},
+     * {@link #getColor}, {@link #getPaint}, and {@link #getValue} are provided as typed alternatives to
+     * the untyped {@link #get} method.
+     * <p>
+     * The preferences that are reported by the platform may be dependent on the operating system version,
+     * so applications should not assume that a particular preference is always available.
+     * <p>
+     * The following list contains all preferences that are potentially available on the specified platforms:
```

Comment:
Is it possible that a preference might be stored that isn't in the below list? Or is it guaranteed that the _only_ possible preferences that will ever be stored are those documented here? If the former, I recommend removing the word "all". If the latter, then it is OK as is, if we are comfortable specifying this.

Reply:
I think we should indeed document _all_ possible preferences for the platform toolkit implementations we control. Of course, this means that we need to keep this list in sync with the actual implementation.

Reply:
+1

we could put the doc (in markdown format) in the doc-files/platform directory (or create any other subdirectory, see JDK-8309749)

Reply:
I think Michael meant that the javadoc-generated API docs in this file need to be kept up to date. If we care going to say _all_, then yes it will. I don't see a need for any other doc related to this.